plugin-k8s/src/lib.rs

```diff
@@ -84,69 +104,153 @@ impl AlumetPlugin for K8sPlugin {
+                        match path.extension() {
+                            None => {
+                                // Case of no extension found --> I will not find cpu.stat file
+                                return;
+                            }
+                            Some(os_str) => match os_str.to_str() {
+                                Some("slice") => {
+                                    // Case of .slice found --> I will find cpu.stat file
+                                    log::debug!(".slice extension found, will continue");
+                                }
+                                _ => {
+                                    // Case of an other extension than .slice is found --> I will not find cpu.stat file
+                                    return;
+                                }
+                            },
+                        };
+                        if let Some(pod_uid) = path.file_name() {
+                            let pod_uid = match pod_uid.to_str() {
+                                Some(uid_str) => uid_str,
+                                None => {
+                                    log::error!("Unable to transform to str");
+                                    return;
+                                }
+                            };
                             // We open a File Descriptor to the newly created file
                             let mut path_cpu = path.clone();
+                            let full_name_to_seek = pod_uid.strip_suffix(".slice").unwrap_or(&pod_uid);
+                            let parts: Vec<&str> = full_name_to_seek.split("pod").collect();
+                            let name_to_seek_raw = *(parts.last().unwrap_or(&full_name_to_seek));
+                            let uid_raw = parts.last().unwrap_or(&"No UID found");
+                            let uid = format!("pod{}", uid_raw);
+                            let name_to_seek = name_to_seek_raw.replace("_", "-");
+                            // let (name, ns) = cgroup_v2::get_pod_name(name_to_seek.to_owned());
+                            let hostname_ostring = gethostname();
+                            let hostname = hostname_ostring
+                                .to_str()
+                                .context("Invalid UTF-8 in Hostname")
+                                .unwrap_or("")
+                                .to_string();
+                            let rt = match tokio::runtime::Builder::new_current_thread().enable_all().build() {
+                                Ok(rt_ok) => rt_ok,
+                                Err(err) => {
+                                    log::error!("Runtime failed to build and returned an error: {}", err);
+                                    return;
+                                }
+                            };
+                            let (name, namespace, node) = match rt.block_on(async {
```

Comment:
Are you aware that this blocks the current thread (the notify thread) until the K8S `get_pod_name` finishes?
It's fine for a first version, but could we imagine something better for the next one?

Reply:
Do you know how to improve it ? Currently I have no idea

Reply:
It should be possible to:
1. put (name, namespace, node) and the code that uses these variables inside the async task
2. get the `Handle` to the async runtime created by Alumet and spawn the task on it
3. let the task run in the background without blocking (i.e. "spawn and forget")

Do you want to create an issue sot that we can do it for a future version?

Reply:
Oh yes I want, it's a very important one. I copy your message in the issue, feel free to add/ modify it